packages/fxa-auth-server/scripts/recorded-future/check-and-reset.ts

```diff
@@ -102,122 +123,169 @@ Limit: ${resultsPerPageLimit}
-        accountLogins.push(identity);
-        statsd.increment(
-          'recorded-future.identity.credentials-search.results.account-emails'
-        );
-      } else {
-        statsd.increment(
-          'recorded-future.identity.credentials-search.results.2fa-accounts'
-        );
-      }
-    }
+  if (has2faCount) {
+    statsd.increment(
+      'recorded-future.identity.credentials-search.results.2fa-accounts',
+      has2faCount
+    );
   }
 
-  if (accountLogins.length === 0) {
-    console.log('No account emails found.');
-    return 0;
+  if (accounts.size === 0) {
+    return [];
   }
 
+  const accountsByLogin: Record<
+    SearchResultIdentity['login'],
+    Awaited<ReturnType<ReturnType<typeof createFindAccountFn>>>
+  > = Array.from(accounts.entries()).reduce((acc, val) => {
+    acc[val[0].login] = val[1];
+    return acc;
+  }, {});
+
   const credentialsLookupFn = createCredentialsLookupFn(client);
-  const cleartextCredentials = await credentialsLookupFn(accountLogins, {
-    first_downloaded_gte: firstDownloadedDateIsoString,
-  });
+  const cleartextCredentials = await credentialsLookupFn(
+    Array.from(accounts.keys()),
+    {
+      first_downloaded_gte: firstDownloadedDate,
+    }
+  );
   statsd.increment(
     'recorded-future.identity.credentials-lookup.cleartext-total',
     cleartextCredentials.length
   );
 
+  const db = await getAuthDb();
   const verifyPassword = createVerifyPasswordFn(
     Password,
-    authDb.checkPassword.bind(authDb)
+    db.checkPassword.bind(db)
   );
 
-  const defaultConcurrency = 100;
-  const concurrency = program.concurrency || defaultConcurrency;
-  const queue = new PQueue({
-    concurrency,
-    interval: 1000,
-    intervalCap: concurrency,
-  });
-
   for (const foundCredentials of cleartextCredentials) {
-    await queue.onSizeLessThan(concurrency * 5);
+    const acct = accountsByLogin[foundCredentials.subject as string];
+    const passwordMatched = await verifyPassword(foundCredentials, acct);
+
+    if (passwordMatched) {
+      accountsToReset.push(acct);
+      statsd.increment(
+        'recorded-future.identity.credentials-lookup.password-match'
+      );
+    }
+  }
+
+  return accountsToReset;
+}
+
+async function getAccountsByLogin(logins: SearchResultIdentity[]) {
+  const findAcct = await getFindAccountFn();
+  const hasTotp2FA = await getHasTotp2faFn();
+  const accounts: Map<
+    SearchResultIdentity,
+    Awaited<ReturnType<ReturnType<typeof createFindAccountFn>>>
+  > = new Map();
+  let has2faCount = 0;
 
-    queue.add(async () => {
-      const acct = accounts[foundCredentials.subject as string];
-      const passwordMatched = await verifyPassword(foundCredentials, acct);
+  for (const x of logins) {
+    const acct = await findAcct(x.login);
 
-      if (passwordMatched) {
-        // @TODO maybe reset and send email
-        statsd.increment(
-          'recorded-future.identity.credentials-lookup.password-match'
-        );
+    if (acct) {
```

Comment:
Should we log anything if no accounts are found?

Reply:
Yeah, I'm adding more logging in the next PR.